### PR TITLE
SoundWire: remove warnings and useless casts

### PR DESCRIPTION
**Pull request by plbossart**
minor cleanups done on sdw/sdca-integration branch but can go upstream without waiting.

### CODE REVIEW

File: sound/soc/codecs/rt711-sdw.c [hunk 1]
```diff
@@ -379,15 +380,15 @@ static int rt711_read_prop(struct sdw_slave *slave)
 	if (!prop->sink_dpn_prop)
 		return -ENOMEM;
 
-	i = 0;
+	j = 0;
```
Comment (lyakh) [inline sound/soc/codecs/rt711-sdw.c:383]: this one is a false positive, `i` is clearly not needed any more, I wouldn't do this. I would just leave this as is, but if we're really keen on pleasing cppcheck, we could initialise `i = -1;` and then `i++`; in the beginning of the loop, then it should be clear, that `i` *is* used after assignment, but I wouldn't bother frankly.

**Review by lyakh — CHANGES_REQUESTED** [review]

File: sound/soc/codecs/rt711-sdw.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/codecs/rt711-sdw.c:383]: Yes it's a false positive but cppcheck found quite a few errors in our code, and it helps to reduce the number of warnings.

this is how it's been fixed for other drivers, so I just added the same recipe for consistency.

Comment (lyakh) [reply sound/soc/codecs/rt711-sdw.c:383]: /me really dislikes making the code less natural for humans to please "intelligent" parsers :-/

Comment (kv2019i) [reply sound/soc/codecs/rt711-sdw.c:383]: This is a bit iffy, but we do the same sometimes for performance, so ok for me.

**Review by kv2019i — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Pull request merged**